src/core/clipboardmanager.cpp

```diff
@@ -0,0 +1,163 @@
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "clipboardmanager.h"
+#include "featureutils.h"
+
+#include <QApplication>
+#include <QDomDocument>
+#include <qgsfeature.h>
+#include <qgsvectorlayer.h>
+
+ClipboardManager::ClipboardManager( QObject *parent )
+  : QObject( parent )
+  , mClipboard( QApplication::clipboard() )
+{
+  connect( QApplication::clipboard(), &QClipboard::dataChanged, this, &ClipboardManager::dataChanged );
+  dataChanged();
+}
+
+void ClipboardManager::dataChanged()
+{
+  if ( mSkipDataChanged )
+  {
+    mSkipDataChanged = false;
+    return;
+  }
+
+  mHasNativeFeature = false;
+  mNativeFeature = QgsFeature();
+
+  bool holdsFeature = false;
+  const QMimeData *mimeData = mClipboard->mimeData();
+  if ( mimeData->hasHtml() )
+  {
+    QDomDocument doc;
+    doc.setContent( mimeData->html() );
+    const QDomNodeList nodes = doc.elementsByTagName( QStringLiteral( "table" ) );
+    if ( !nodes.isEmpty() )
+    {
+      const QDomElement table = nodes.at( 0 ).toElement();
+      if ( table.hasAttribute( QStringLiteral( "qfield" ) ) )
+      {
+        holdsFeature = true;
+      }
+    }
+  }
+
+  if ( mHoldsFeature != holdsFeature )
+  {
+    mHoldsFeature = holdsFeature;
+    emit holdsFeatureChanged();
+  }
+}
+
+void ClipboardManager::copyFeatureToClipboard( QgsVectorLayer *layer, QgsFeatureId fid, bool includeGeometry )
+{
+  if ( layer )
+  {
+    const QgsFeature feature = layer->getFeature( fid );
+    copyFeatureToClipboard( feature, includeGeometry );
+  }
+}
+
+void ClipboardManager::copyFeatureToClipboard( const QgsFeature &feature, bool includeGeometry )
+{
+  QStringList textLines;
+  QStringList htmlLines;
+
+  if ( includeGeometry )
+  {
+    textLines << QStringLiteral( "%1: %2" ).arg( QStringLiteral( "%1:" ).arg( tr( "Geometry" ) ), feature.geometry().asWkt() );
+    htmlLines << QStringLiteral( "<td geometry=\"1\">%1</td><td>%2</td>" ).arg( QStringLiteral( "%1:" ).arg( tr( "Geometry" ) ), feature.geometry().asWkt().toHtmlEscaped() );
+  }
+
+  const QgsFields fields = feature.fields();
+  const QgsAttributes attributes = feature.attributes();
+  for ( int i = 0; i < fields.count(); i++ )
+  {
+    textLines << QStringLiteral( "%1: %2" ).arg( fields[i].displayName(), attributes[i].toString() );
+    htmlLines << QStringLiteral( "<td>%1</td><td>%2</td>" ).arg( fields[i].displayName(), attributes[i].toString().toHtmlEscaped() );
+  }
+
+  QMimeData *mimeData = new QMimeData();
+  mimeData->setText( textLines.join( '\n' ) );
+  mimeData->setHtml( QStringLiteral( "<!DOCTYPE html><html><head><meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\"/></head><body><table border=\"1\" qfield=\"1\"><tr>" ) + htmlLines.join( QStringLiteral( "</tr><tr>" ) ) + QStringLiteral( "</tr></table></body></html>" ) );
```

Comment:
```suggestion
  mimeData->setHtml( QStringLiteral( R""""(
<!DOCTYPE html>
<html>
    <head>
        <meta http-equiv=\"content-type\" content=\"text/html; charset=utf-8\"/>
    </head>
    <body>
        <table border=\"1\" qfield=\"1\">
            <tr>
            %1
            </tr>
        </table>
    </body>
</html>
    )""""
  )).arg( htmlLines.join( QStringLiteral( "</tr><tr>" ) ) );
```

I think this slightly improves the readability :)

Reply:
@suricactus , your desire is my command :wink: